### PR TITLE
Feature/cleanup for 8

### PR DESCRIPTION
**Pull request by kudashevs**
Hello,

I'd like to propose some improvements:
- add a support for the `float` type
- add a type to the `fileName` field
- add a `static` return type to all methods that return $this (because it's PHP 8 only package)
- add explicit union type to the methods that output stored values (the union type is `array|string|float|int|null`)
- inline `startsWith` method (because it's PHP 8 only package and we can use a built-in function)

### CODE REVIEW

File: src/Valuestore.php [hunk 1]
```diff
@@ -48,12 +47,12 @@ protected function setFileName(string $fileName)
     /**
      * Put a value in the store.
      *
-     * @param string|array    $name
-     * @param string|int|null $value
+     * @param array|string                $name
+     * @param array|string|float|int|null $value
```
Comment (freekmurze) [inline src/Valuestore.php:51]: Can't we just typehint mixed? Would look nice, and we don't need the doc block anymore.

File: src/Valuestore.php [hunk 2]
```diff
@@ -48,12 +47,12 @@ protected function setFileName(string $fileName)
     /**
      * Put a value in the store.
      *
-     * @param string|array    $name
-     * @param string|int|null $value
+     * @param array|string                $name
```
Comment (freekmurze) [inline src/Valuestore.php:50]: Can't we just type this instead of the docblock?

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/Valuestore.php (hunk 1 above)
Comment (kudashevs) [reply src/Valuestore.php:51]: I kind of agree, however, the `mixed` type is wider than the union type we need here (It includes objects and resources. And objects and resources cannot be proceed by the package). So, I see to options here:
- leave it as it is with a type-hint in the docblock
- add this union type as a type-hint to the $value parameter, but it's too wordy though (however I would prefer this approach because it validates input and leaves no chance for misuse)

What do you think?

File: src/Valuestore.php (hunk 2 above)
Comment (kudashevs) [reply src/Valuestore.php:50]: I've added the type-hint to the parameter.

File: src/Valuestore.php (hunk 1 above)
Comment (kudashevs) [reply src/Valuestore.php]: UP: I've uploaded my take on how it could look like with long wordy union types. But I would prefer this approach. It's explicit and it follows the last trends (to use native types and remove superfluous PHP annotations).

File: src/Valuestore.php [hunk 3]
```diff
@@ -235,32 +235,34 @@ public function pull(string $name): array|string|float|int|null
 
     /**
      * Increment a value from the store.
-     *
-     * @param string $name
-     * @param int    $by
-     *
-     * @return string|int|null
```
Comment (kudashevs) [inline src/Valuestore.php]: Here I wasn't really sure about the intention. To me it's too much magic to allow incrementing strings.
